zerver/lib/outgoing_webhook.py

```diff
@@ -81,11 +104,15 @@ def send_response_message(bot_id, message, response_message_content):
 
 def succeed_with_message(event, success_message):
     # type: (Dict[str, Any], Text) -> None
+    if success_message is None:
+        return
     success_message = "Success! " + success_message
     send_response_message(event['user_profile_id'], event['message'], success_message)
 
 def fail_with_message(event, failure_message):
     # type: (Dict[str, Any], Text) -> None
+    if failure_message is None:
+        return
```

Comment:
I don't understand why these are checking for `None` and the code is in `process_success` sometimes returns `""`.  This needs to be consistent.  I would use `None` here and modify the `process_success` functions to return `None`, so you're only handling `""` at the boundary interface.